apps/prairielearn/src/lib/score-upload.js

```diff
@@ -9,372 +9,387 @@ const manualGrading = require('./manualGrading');
+    const csvStream = streamifier.createReadStream(csvFile.buffer, {
+      encoding: 'utf8',
     });
-
-    const assessment_label = result.rows[0].assessment_label;
-    const course_instance_id = result.rows[0].course_instance_id;
-    const course_id = result.rows[0].course_id;
-
-    const serverJob = await createServerJob({
-      courseId: course_id,
-      courseInstanceId: course_instance_id,
-      assessmentId: assessment_id,
-      userId: user_id,
-      authnUserId: authn_user_id,
-      type: 'upload_instance_question_scores',
-      description: 'Upload question scores for ' + assessment_label,
+    const csvConverter = csvtojson({
+      colParser: {
+        instance: 'number',
+        score_perc: 'number',
+        points: 'number',
+        manual_score_perc: 'number',
+        manual_points: 'number',
+        auto_score_perc: 'number',
+        auto_points: 'number',
+        submission_id: 'number',
+      },
+      maxRowLength: 10000,
     });
 
-    serverJob.executeInBackground(async (job) => {
-      job.info('Uploading question scores for ' + assessment_label);
-
-      // accumulate output lines in the "output" variable and actually
-      // output put them in blocks, to avoid spamming the updates
-      let output = null;
-      let outputCount = 0;
-      let outputThreshold = 100;
-
-      let successCount = 0;
-      let errorCount = 0;
-
-      job.info(`Parsing uploaded CSV file "${csvFile.originalname}" (${csvFile.size} bytes)`);
-      const csvStream = streamifier.createReadStream(csvFile.buffer, {
-        encoding: 'utf8',
-      });
-      const csvConverter = csvtojson({
-        colParser: {
-          instance: 'number',
-          score_perc: 'number',
-          points: 'number',
-          manual_score_perc: 'number',
-          manual_points: 'number',
-          auto_score_perc: 'number',
-          auto_points: 'number',
-          submission_id: 'number',
-        },
-        maxRowLength: 10000,
-      });
-
-      try {
-        await csvConverter.fromStream(csvStream).subscribe(async (json, number) => {
-          // Replace all keys with their lower-case values
-          json = _.mapKeys(json, (_v, k) => k.toLowerCase());
-          const msg = `Processing CSV record ${number}: ${JSON.stringify(json)}`;
-          if (output == null) {
-            output = msg;
-          } else {
-            output += '\n' + msg;
-          }
-          try {
-            await module.exports._updateInstanceQuestionFromJson(
-              json,
-              assessment_id,
-              authn_user_id,
-            );
+    try {
+      await csvConverter.fromStream(csvStream).subscribe(async (json, number) => {
+        // Replace all keys with their lower-case values
+        json = _.mapKeys(json, (_v, k) => k.toLowerCase());
+        try {
+          if (await _updateInstanceQuestionFromJson(json, assessment_id, authn_user_id)) {
```

Comment:
As long as you're making changes, consider dropping the leading `_` from function names; this is a pattern that we use almost nowhere else. I think it's meant to indicate a "private" function. This would have been marginally more important when we stuck all functions in `module.exports = { ... }`, but now that we're `export`ing just the ones that are actually meant to be used from elsewhere, I don't see a benefit in keeping the `_`.

Reply:
Done in a253a4bdc07efdb86272d8b198bcd2da302b73dd.